From b06868f156069fdd303f387f00a55cc2a69df1e1 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Mon, 20 Feb 2023 19:01:18 -0500
Subject: ucrt: Replace "msvcrt" with "ucrtbase"

Replace "-lmsvcrt" with "-lucrtbase" in base/linking.jl
Set CRTDLL_BASENAME to "ucrtbase".
Replace libmsvcrt.a with libucrtbase.a
---
 Makefile        | 2 +-
 base/linking.jl | 2 +-
 src/dlload.c    | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 0d85d19567..b40ee6045f 100644
--- a/Makefile
+++ b/Makefile
@@ -276,7 +276,7 @@ endif
 	# These are files from mingw32 and required for creating shared libraries like our caches.
 	-$(INSTALL_M) $(build_libdir)/libgcc_s.a $(DESTDIR)$(libdir)/
 	-$(INSTALL_M) $(build_libdir)/libgcc.a $(DESTDIR)$(libdir)/
-	-$(INSTALL_M) $(build_libdir)/libmsvcrt.a $(DESTDIR)$(libdir)/
+	-$(INSTALL_M) $(build_libdir)/libucrtbase.a $(DESTDIR)$(libdir)/
 else
 
 # Copy over .dSYM directories directly for Darwin
diff --git a/base/linking.jl b/base/linking.jl
index 18cf128b34..0c8d47bfed 100644
--- a/base/linking.jl
+++ b/base/linking.jl
@@ -155,7 +155,7 @@ function link_image_cmd(path, out)
     SHLIBDIR = "-L$(shlibdir())"
     LIBS = is_debug() ? ("-ljulia-debug", "-ljulia-internal-debug") : ("-ljulia", "-ljulia-internal")
     @static if Sys.iswindows()
-        LIBS = (LIBS..., "-lopenlibm", "-lssp", "-lgcc_s", "-lgcc", "-lmsvcrt")
+        LIBS = (LIBS..., "-lopenlibm", "-lssp", "-lgcc_s", "-lgcc", "-lucrtbase")
     end
 
     V = VERBOSE[] ? "--verbose" : ""
diff --git a/src/dlload.c b/src/dlload.c
index 9f4e8be299..cd584461a1 100644
--- a/src/dlload.c
+++ b/src/dlload.c
@@ -60,7 +60,7 @@ static int endswith_extension(const char *path) JL_NOTSAFEPOINT
 }
 
 #ifdef _OS_WINDOWS_
-#define CRTDLL_BASENAME "msvcrt"
+#define CRTDLL_BASENAME "ucrtbase"
 
 JL_DLLEXPORT const char *jl_crtdll_basename = CRTDLL_BASENAME;
 const char *jl_crtdll_name = CRTDLL_BASENAME ".dll";
-- 
